cmd/topicctl/subcmd/rebalance.go

```diff
@@ -151,55 +159,97 @@ func rebalanceRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	// iterate through each topic config and initiate rebalance
-	topicConfigs := []config.TopicConfig{}
-	topicErrorDict := make(map[string]error)
-	for _, topicFile := range topicFiles {
-		// do not consider invalid topic yaml files for rebalance
-		topicConfigs, err = config.LoadTopicsFile(topicFile)
+	tmpDir := ""
+	existingConfigFiles := []string{}
+	if rebalanceConfig.generateConfig {
+		// make set of existing files
+		err := processTopicFiles(topicFiles, func(topicConfig config.TopicConfig, topicFile string) error {
+			existingConfigFiles = append(existingConfigFiles, topicConfig.Meta.Name)
+			return nil
+		})
 		if err != nil {
-			log.Errorf("Invalid topic yaml file: %s", topicFile)
-			continue
+			return err
 		}
 
-		for _, topicConfig := range topicConfigs {
-			// topic config should be consistent with the cluster config
-			if err := config.CheckConsistency(topicConfig.Meta, clusterConfig); err != nil {
-				log.Errorf("topic file: %s inconsistent with cluster: %s", topicFile, clusterConfigPath)
-				continue
-			}
+		// create temp dir
+		tmpDir = rebalanceConfig.pathPrefix
+		if rebalanceConfig.pathPrefix[len(rebalanceConfig.pathPrefix) - 1] != '/' {
+			tmpDir += "/"
+		}
+		tmpDir += "tmp/"
```

Comment:
I wonder about using tmp/ to write the bootstrapped topic configs. The flag name is --generate-config so as a naive user, I think my expectation would be that those files would be created and left but you are deleting them after processing. Could you elaborate on this design, why you've opted to use a tmp dir that is cleaned up afterward?